src/griddyn/loads/aggregateLoad.cpp

```diff
@@ -114,14 +114,16 @@ void aggregateLoad::pFlowObjectInitializeA (coreTime time0, std::uint32_t flags)
         // do a first pass of loading
         double rem = 1.0;
 
-        setP (((aggregateLoad *)sLoad)->getP ());  // so this composite load function has direct access to the
+        aggregateLoad* sLoadRepr = dynamic_cast<aggregateLoad*>(sLoad);
+        assert(sLoadRepr != nullptr);
```

Comment:
Need to look at zipLoad vs aggregateLode, to see if this can actually succeed

Reply:
aggregate load builds on zipload, so all aggregate loads are ziploads.